### DISCUSSION TITLE
Get Messages API only fetching the last 20 Messages in the Conversation

### DISCUSSION THREAD

**Discussion by devrulabs** [chatwoot/chatwoot#2496] [Q&A] [answered]
https://app.chatwoot.com/api/v1/accounts/{account_id}/conversations/{converstion_id}/messages

This API only Returns the last 20 conversations for a given Conversation ID. How to get the entire conversation without the limit of last 20 Conversation ?

**Comment by sojan-official** [comment, marked as answer]
at the moment the API only supports getting 20 messages at a time. you can paginate by providing the parameter `before`  which accepts a message-id for value.

**Reply by devrulabs** [reply]
Thanks @sojan-official This helps

**Reply by m-ret** [reply]
> at the moment the API only supports getting 20 messages at a time. you can paginate by providing the parameter `before` which accepts a message-id for value.

Where is this documented?

**Comment by immediando** [comment]
any news about getting the entire conversation without the limit of last 20 Conversation ?

**Comment by panaras10** [comment]
Can we fix this please? Seems like an easy one and pretty obvious.

**Comment by juani-gk** [comment]
it would be awesome to be able to get more messages in cloud version
